exec/java-exec/src/test/java/org/apache/drill/exec/memory/TestAllocators.java

```diff
@@ -63,6 +69,53 @@
   private final static String planFile="/physical_allocator_test.json";
 
   @Test
+  public void testDrillBufReadIndex() throws Exception {
```

Comment:
Changed the method name and added a doc to describe the scenario where the failing cases would occur.

Thanks for reviewing.